### ISSUE TITLE
DefaultHttpClient.proxy ignores connection pool limits

### ISSUE THREAD

**Issue by bwertman-rc** [micronaut-projects/micronaut-core#7028]
### Expected Behavior

I would like to limit the active concurrent connections to an upstream server when proxying requests using a `DefaultHttpClient`, preferably by setting a connection limit in its configuration.

Alternately, I could use the normal `exchange` methods on `DefaultHttpClient` (which presumably respect the configured limits) if the proxied request rewriting was extracted into a separate class for reusability.

### Actual Behaviour

I tried enabling the client's connection pool and setting a limit, but during testing I found it wasn't being respected. Stepping though the code in a debugger, it seems that `poolMap` isn't accessed at all.

### Steps To Reproduce

Configure a HttpClientConfiguration's connection pool to enable it and set a limit on the number of connections.

```
HttpClientConfiguration config = new DefaultHttpClientConfiguration();
ConnectionPoolConfiguration conn = config.getConnectionPoolConfiguration();
conn.setEnabled(true);
conn.setMaxConnections(maxConcurrentRequests);
```

Use that configuration to create a proxy client

```
ProxyHttpClient proxyClient = ProxyHttpClient.create(url, config);
```

Proxy requests through the client, and see more connections than the configured pool size.

### Environment Information

- Operating system: Linux 
- Java version: OpenJDK 17

### Version

3.3.4

**Comment by yawkat** [comment]
yes the proxy client doesn't participate in connection pooling. pooling for the http client in general needs some work

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?

**Comment by graemerocher** [comment]
Triage: the report is consistent with the current proxy-client path: `ProxyHttpClient` does not participate in the normal `DefaultHttpClient` connection-pool accounting, as the maintainer noted. Suggested fix: route proxied exchanges through the same connection manager/pool or give the proxy client an explicit bounded pool, then add a concurrency test asserting the configured maximum. Please confirm behavior and configuration names on a current release.
